APMrover2/Rover.cpp

```diff
@@ -324,6 +327,33 @@ void Rover::update_mission(void)
     }
 }
 
+#if OSD_ENABLED == ENABLED
+void Rover::publish_osd_info()
+{
+    AP_OSD::NavInfo nav_info;
+    bool navmode = (control_mode == &mode_auto) || (control_mode == &mode_rtl) || (control_mode == &mode_smartrtl) || (control_mode == &mode_guided);
```

Comment:
Explicitly calling modes out isn't a great thing to do.  Using methods on the current flight mode is the appropriate thing to do.

https://github.com/ardupilot/ardupilot/blob/master/APMrover2/GCS_Mavlink.cpp#L100 can be used as a reference for this.

Reply:
fixed per Randy's suggestions